test/features/push_notification/presentation/websocket/web_socket_queue_handler_test.dart

```diff
@@ -0,0 +1,100 @@
+
+import 'package:core/utils/app_logger.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jmap_dart_client/jmap/core/state.dart';
+import 'package:tmail_ui_user/features/push_notification/presentation/websocket/web_socket_message.dart';
+import 'package:tmail_ui_user/features/push_notification/presentation/websocket/web_socket_queue_handler.dart';
+
+import '../../../../fixtures/account_fixtures.dart';
+import '../../../../fixtures/session_fixtures.dart';
+
+void main() {
+  group('WebSocketQueueHandler::test', () {
```

Comment:
Please structure the test for this class:
1. Basic Operations:
    - Messages are processed in the correct order
    - Duplicate messages are handled correctly
    - ...
2. Concurrent Operations:
    - Messages can be safely added while processing is ongoing
    - Order is maintained even under high concurrency
    - ...
3. Error Handling:
    - Should continue processing after message failure
    - Should add message after message failure
    - ...
4. Stress testing:
    - Can handle large bursts of messages
    - should handle interleaved slow and fast messages
    - ...